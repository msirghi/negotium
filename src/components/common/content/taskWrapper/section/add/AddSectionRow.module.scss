.adrTitleContainer {
  opacity: 0;
  width: 100%;
  height: 20px;
  border-bottom: 1px solid lightgray;
  text-align: center;
  position: relative;
  transition: 0.5s opacity ease-in-out;

  &:hover {
    cursor: pointer;
    opacity: 1;
  }

  .adrTitle {
    position: absolute;
    left: 50%;
    top: -5px;
    transform: translate(-50%, 0);
    padding: 10px 10px;
    background-color: #fff;
  }
}

.buttonRow {
  margin-top: .5rem;
}
