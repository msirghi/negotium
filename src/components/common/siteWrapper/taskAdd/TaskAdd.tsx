import AddIcon from '@mui/icons-material/Add';
import { IconButton, Tooltip } from '@mui/material';
import { useAtom } from 'jotai';
import { TaskAddDialog } from './dialog/TaskAddDialog';
import useTranslation from 'next-translate/useTranslation';
import { showTaskDialog } from '../../../../atoms/taskDialogOpened/taskDialogOpened.atom';

export const TaskAdd = () => {
  const [taskDialogOpened, setTaskDialogOpened] = useAtom(showTaskDialog);
  const { t } = useTranslation('common');

  const toggleDialog = () => setTaskDialogOpened(!taskDialogOpened);

  return (
    <>
      <Tooltip title={t('inputPlaceholders.addTask')}>
        <IconButton color="inherit" onClick={toggleDialog}>
          <AddIcon />
        </IconButton>
      </Tooltip>

      <TaskAddDialog open={taskDialogOpened} handleClose={toggleDialog} />
    </>
  );
};
