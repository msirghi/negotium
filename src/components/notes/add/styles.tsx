import { makeStyles } from '@mui/styles';

export const useNotesAddInputStyles = makeStyles({
  container: {
    width: '100%',
    display: 'flex',
    alignItems: 'center',
    flexDirection: 'column',
  },
  inputToggler: {
    width: '33%',
    borderColor: '#e0e0e0',
    border: '1px solid',
    borderRadius: 5,
    padding: 5,
  },
  input: {
    width: '100%',
    marginLeft: 10
  },
});
