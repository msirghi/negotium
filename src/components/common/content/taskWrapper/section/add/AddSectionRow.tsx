import styles from './AddSectionRow.module.scss';
import { FC, useState } from 'react';
import { Button, TextField } from '@mui/material';
import { If } from '../../../../utilities/if/If';
import { Row } from '../../../../utilities/row/Row';
import { useTranslation } from 'next-i18next';

type Props = {
  onSectionSave: (title: string, orderNumber: number) => void;
};

export const AddSectionRow: FC<Props> = ({ onSectionSave }) => {
  const [editMode, setEditMode] = useState(false);
  const [fieldValue, setFieldValue] = useState('');
  const { t } = useTranslation('common');

  const onTitleClick = () => {
    setEditMode(true);
  };

  const onCancelClick = () => {
    setEditMode(false);
  };

  const onSave = () => {
    onSectionSave(fieldValue, 1);
    setEditMode(false);
    setFieldValue('');
  };

  return (
    <div>
      <If condition={!editMode}>
        <div
          className={styles.adrTitleContainer}
          onClick={onTitleClick}
          data-testid={'title'}
        >
          <span className={styles.adrTitle}>Add section</span>
        </div>
      </If>
      <If condition={editMode}>
        <TextField
          onChange={(e) => setFieldValue(e.target.value)}
          fullWidth
          size={'small'}
          placeholder={t('inputPlaceholders.nameNewSection')}
          inputProps={{ 'data-testid': 'section-name-field' }}
          sx={{ marginTop: '1rem' }}
        />
        <Row className={styles.buttonRow}>
          <Button
            onClick={onSave}
            disabled={!fieldValue}
            data-testid={'save-button'}
            color={'primary'}
            variant={'contained'}
            size={'small'}
          >
            {t('buttonLabels.addSection')}
          </Button>
          <Button
            onClick={onCancelClick}
            size={'small'}
            data-testid={'cancel-button'}
          >
            {t('buttonLabels.cancel')}
          </Button>
        </Row>
      </If>
    </div>
  );
};
