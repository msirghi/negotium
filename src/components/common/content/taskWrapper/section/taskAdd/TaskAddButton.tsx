import { Box, Button, ButtonProps, styled } from '@mui/material';
import AddIcon from '@mui/icons-material/Add';
import { FC, MutableRefObject, useRef, useState } from 'react';
import { If } from '../../../../utilities/if/If';
import { EditForm } from './editForm/EditForm';
import { makeStyles } from '@mui/styles';
import { Nullable } from '../../../../../../common/types/common.types';
import { useIsMobile } from '../../../../../../common/hooks/common/useIsMobile';

const CustomButton = styled(Button)<ButtonProps>(() => ({
  color: 'grey',
  width: '100%'
}));

const useStyles = makeStyles({
  cancelButton: {
    marginLeft: 10,
  },
});

type Props = {
  onTaskAdd: (title: string, date: Nullable<Date>) => void;
  defaultDate?: Date;
};

export const TaskAddButton: FC<Props> = ({ onTaskAdd, defaultDate }) => {
  const [editMode, setEditMode] = useState(false);
  const [fieldValue, setFieldValue] = useState('');
  const classes = useStyles();
  const selectedDate = useRef<Nullable<Date>>(
    defaultDate || null
  ) as MutableRefObject<Nullable<Date>>;
  const isMobile = useIsMobile();

  const toggleEditMode = () => {
    setEditMode(!editMode);
  };

  const onCancelClick = () => {
    setEditMode(false);
    setFieldValue('');
  };

  const onSave = () => {
    onTaskAdd(fieldValue, selectedDate.current);
    onDateSelect(null);
    setEditMode(false);
    setFieldValue('');
  };

  const onDateSelect = (date: Nullable<Date>) => {
    selectedDate.current = date;
  };

  return (
    <>
      <If condition={!editMode}>
        <CustomButton
          data-testid={'tab-add-button'}
          startIcon={<AddIcon />}
          disableRipple
          onClick={toggleEditMode}
        >
          Add
        </CustomButton>
      </If>
      <If condition={editMode}>
        <Box sx={{ marginTop: '1rem' }}>
          <form onSubmit={onSave}>
            <EditForm
              defaultDate={defaultDate}
              fieldValue={fieldValue}
              setFieldValue={setFieldValue}
              onDateSelect={onDateSelect}
            />
            <Box sx={{ marginTop: '.5rem' }}>
              <Button
                variant={'contained'}
                data-testid={'tab-submit-button'}
                disabled={!fieldValue}
                onClick={onSave}
                size={'small'}
              >
                Add task
              </Button>
              <Button
                className={classes.cancelButton}
                data-testid={'tab-cancel-button'}
                onClick={onCancelClick}
                size={'small'}
              >
                Cancel
              </Button>
            </Box>
          </form>
        </Box>
      </If>
    </>
  );
};
