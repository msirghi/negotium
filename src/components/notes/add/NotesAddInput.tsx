import { FC, useState } from 'react';
import { Box } from '@mui/system';
import { TextField } from '@mui/material';
import { useNotesAddInputStyles } from './styles';
import { NotesAddForm } from './form/NotesAddForm';
import { If } from '../../common/utilities/if/If';
import { BorderlessInput } from '../../common/form/input/borderlessInput/BorderlessInput';

export const NotesAddInput: FC = () => {
  const classes = useNotesAddInputStyles();

  const [editFormOpen, setEditFormOpen] = useState(false);

  const openEditForm = () => setEditFormOpen(true);

  const closeEditForm = () => setEditFormOpen(false);

  return (
    <Box className={classes.container}>
      <If condition={!editFormOpen}>
        <Box className={classes.inputToggler}>
          <BorderlessInput
            variant={'standard'}
            onFocus={openEditForm}
            className={classes.input}
            placeholder={'Add a note...'}
          />
        </Box>
      </If>
      <If condition={editFormOpen}>
        <NotesAddForm onClose={closeEditForm} />
      </If>
    </Box>
  );
};
