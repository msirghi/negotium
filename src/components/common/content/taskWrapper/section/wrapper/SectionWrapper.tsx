import { FC } from 'react';
import { Section, Task } from '../../../../../../common/types/tasks.types';
import styles from './SectionWrapper.module.scss';
import { TaskItem } from '../../taskItem/TaskItem';
import { Divider } from '@mui/material';
import { TaskAddButton } from '../taskAdd/TaskAddButton';
import { DragDropContext, Droppable } from 'react-beautiful-dnd';
import { Nullable } from '../../../../../../common/types/common.types';
import { EditableTitle } from '../../../pageTitle/editableTitle/EditableTitle';
import SmoothList from 'react-smooth-list';

type Props = {
  sectionId: string;
  title: string;
  tasks: Task[];
  onTaskAdd: (title: string, date: Nullable<Date>, sectionId: string) => void;
  markAsDone: (id: string) => void;
  onTaskSelect: (task: Task) => void;
  onSectionUpdate: (
    title: Section['title'],
    sectionId: Props['sectionId']
  ) => void;
  onSectionRemove: (sectionId: Section['id']) => void;
};

export const SectionWrapper: FC<Props> = ({
  title,
  tasks,
  onTaskAdd,
  sectionId,
  markAsDone,
  onTaskSelect,
  onSectionUpdate,
  onSectionRemove,
}) => {
  const handleTaskAdd = () => {
    return (title: string, date: Nullable<Date>) =>
      onTaskAdd(title, date, sectionId);
  };

  const handleSectionUpdate = () => {
    return (newTitle: string) => onSectionUpdate(newTitle, sectionId);
  };

  const handleSectionRemove = () => {
    return onSectionRemove(sectionId);
  };

  return (
    <div className={styles.swContainer}>
      <EditableTitle
        title={title}
        className={styles.swTitle}
        editableOptions={{
          onSave: handleSectionUpdate(),
          title,
          inputFontSize: 16,
          onRemove: handleSectionRemove,
        }}
      >
        {title}
      </EditableTitle>
      <Divider />
      <div className={styles.swTaskContainer}>
        <DragDropContext onDragEnd={() => {}}>
          <Droppable droppableId={`section-tasks=${sectionId}`}>
            {(provided) => (
              <div ref={provided.innerRef} {...provided.droppableProps}>
                <SmoothList>
                  {tasks.map((task, idx) => {
                    return (
                      <TaskItem
                        task={task}
                        key={task.id}
                        onTaskSelect={onTaskSelect}
                        markAsDone={markAsDone}
                        index={idx}
                      />
                    );
                  })}
                </SmoothList>
                {provided.placeholder}
              </div>
            )}
          </Droppable>
        </DragDropContext>
        <TaskAddButton onTaskAdd={handleTaskAdd()} />
      </div>
    </div>
  );
};
