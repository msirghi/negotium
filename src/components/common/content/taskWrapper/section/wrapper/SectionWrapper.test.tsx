import { TasksMock } from '../../../../../../common/tests/mockData/tasks-mock';
import renderer from 'react-test-renderer';
import { SectionWrapper } from './SectionWrapper';
import { mount } from 'enzyme';
import { TaskAddButton } from '../taskAdd/TaskAddButton';
import { act } from '@testing-library/react';
import { AddSectionRow } from '../add/AddSectionRow';
import { MockDndProvider } from '../../../../../../common/tests/TestUtils';

describe('SectionWrapper', () => {
  const defaultProps = {
    sectionId: '1',
    title: 'Title',
    tasks: TasksMock,
    onSectionAdd: jest.fn(),
    onTaskAdd: jest.fn(),
  };

  it('should match the snapshot', () => {
    const tree = renderer.create(
      <MockDndProvider>
        <SectionWrapper {...defaultProps} />
      </MockDndProvider>
    );
    expect(tree).toMatchSnapshot();
  });

  it('should handle new task add', () => {
    const wrapper = mount(
      <MockDndProvider>
        <SectionWrapper {...defaultProps} />
      </MockDndProvider>
    );
    const addButton = wrapper.find(TaskAddButton);
    act(() => {
      addButton.props().onTaskAdd('title', null);
    });
    expect(defaultProps.onTaskAdd).toBeCalled();
  });

  it('should handle new section add', () => {
    const wrapper = mount(
      <MockDndProvider>
        <SectionWrapper {...defaultProps} />
      </MockDndProvider>
    );
    const addSectionRow = wrapper.find(AddSectionRow);

    act(() => {
      addSectionRow.props().onSectionSave('title', 1);
    });
    expect(defaultProps.onSectionAdd).toBeCalled();
  });
});
