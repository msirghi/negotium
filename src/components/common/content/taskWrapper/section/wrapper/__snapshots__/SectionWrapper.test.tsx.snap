// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SectionWrapper should match the snapshot 1`] = `
<div
  className="swContainer"
>
  <div>
    <div
      className="adrTitleContainer"
      data-testid="title"
      onClick={[Function]}
    >
      <span
        className="adrTitle"
      >
        Add section
      </span>
    </div>
  </div>
  <div
    className="swTitle"
  >
    Title
  </div>
  <hr
    className="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
  />
  <div
    className="swTaskContainer"
  >
    <div
      className="tiItem"
      onClick={[Function]}
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "width": "initial",
        }
      }
    >
      <div
        onClick={[Function]}
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "width": "initial",
          }
        }
      >
        <span
          className="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
          onBlur={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={null}
        >
          <input
            className="PrivateSwitchBase-input css-1m9pwf3"
            data-indeterminate={false}
            onChange={[Function]}
            type="checkbox"
          />
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
            data-testid="CheckBoxOutlineBlankIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
            />
          </svg>
        </span>
        <div>
          [{"type":"paragraph","children":[{"text":"new task for 123 "},{"text":""}]}]
        </div>
      </div>
      <div />
    </div>
    <div
      className="tiItem"
      onClick={[Function]}
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "width": "initial",
        }
      }
    >
      <div
        onClick={[Function]}
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "width": "initial",
          }
        }
      >
        <span
          className="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
          onBlur={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={null}
        >
          <input
            className="PrivateSwitchBase-input css-1m9pwf3"
            data-indeterminate={false}
            onChange={[Function]}
            type="checkbox"
          />
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
            data-testid="CheckBoxOutlineBlankIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
            />
          </svg>
        </span>
        <div>
          [{"type":"paragraph","children":[{"text":"new task for 123 "},{"text":""}]}]
        </div>
      </div>
      <div />
    </div>
    <button
      className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root css-1osibp4-MuiButtonBase-root-MuiButton-root"
      data-testid="tab-add-button"
      disabled={false}
      id="tab-add-button"
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="AddIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
          />
        </svg>
      </span>
      buttonLabels.add
    </button>
  </div>
</div>
`;
